### PR TITLE
Add initial run (integration test) framework and initial actix web 3 support

### PR DESCRIPTION
**Pull request by vaporos**

### CODE REVIEW

File: libsplinter/src/rest_api/actix_web_3/api.rs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+#[derive(Debug)]
+pub struct BindAddress {
+    /// The SocketAddr which defines the bound port.
+    pub addr: SocketAddr,
+
+    /// The scheme (such as http) that is running on this port.
+    pub scheme: String,
+}
+
+enum FromThreadMessage {
```
Comment (peterschwarz) [inline libsplinter/src/rest_api/actix_web_3/api.rs:46]: Could this enum just be a `Result`?

Comment (vaporos) [reply libsplinter/src/rest_api/actix_web_3/api.rs:46]: Technically. However, I don't mean this as a Result in meaning, but as the channel API definition. Using enum for things sent over channels lets you name it--FromThreadMessage--and leaves room for sending additional messages over the channel in the future without changing the fundamental channel type. (This is a general reason, not specific to this situation.)

**Review by peterschwarz — DISMISSED** [review]

File: libsplinter/src/rest_api/actix_web_3/api.rs (hunk 1 above)
Comment (vaporos) [reply libsplinter/src/rest_api/actix_web_3/api.rs:46]: This and the loop{} for the channel will be considered prior to feature stabilization.

File: libsplinter/src/run/mod.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
+// Copyright 2018-2020 Cargill Incorporated
```
Comment (agunde406) [inline libsplinter/src/run/mod.rs:1]: 2021

File: libsplinter/src/run/network.rs [hunk 3]
```diff
@@ -0,0 +1,91 @@
+// Copyright 2018-2020 Cargill Incorporated
```
Comment (agunde406) [inline libsplinter/src/run/network.rs:1]: 2021

File: libsplinter/src/run/network.rs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+impl ShutdownHandle for Network {
+    fn signal_shutdown(&mut self) {
+        for node in &mut self.nodes {
+            node.signal_shutdown()
+        }
+    }
+
+    fn wait_for_shutdown(&mut self, timeout: Duration) -> Result<(), InternalError> {
+        // FIXME -f igure out how to implement this wait to respect timeout
```
Comment (agunde406) [inline libsplinter/src/run/network.rs:77]: Should this fix me comment be removed?

File: libsplinter/src/run/node/builder.rs [hunk 5]
```diff
@@ -0,0 +1,128 @@
+// Copyright 2018-2020 Cargill Incorporated
```
Comment (agunde406) [inline libsplinter/src/run/node/builder.rs:1]: 2021

File: libsplinter/src/run/node/mod.rs [hunk 6]
```diff
@@ -0,0 +1,25 @@
+// Copyright 2018-2020 Cargill Incorporated
```
Comment (agunde406) [inline libsplinter/src/run/node/mod.rs:1]: 2021

File: libsplinter/src/run/node/runnable.rs [hunk 7]
```diff
@@ -0,0 +1,107 @@
+// Copyright 2018-2020 Cargill Incorporated
```
Comment (agunde406) [inline libsplinter/src/run/node/runnable.rs:1]: 2021

File: libsplinter/src/run/node/running.rs [hunk 8]
```diff
@@ -0,0 +1,82 @@
+// Copyright 2018-2020 Cargill Incorporated
```
Comment (agunde406) [inline libsplinter/src/run/node/running.rs:1]: 2021

File: libsplinter/src/run/node_config.rs [hunk 9]
```diff
@@ -0,0 +1,23 @@
+// Copyright 2018-2020 Cargill Incorporated
```
Comment (agunde406) [inline libsplinter/src/run/node_config.rs:1]: 2021

**Review by agunde406 — CHANGES_REQUESTED** [review]

File: libsplinter/src/run/network.rs (hunk 4 above)
Comment (vaporos) [reply libsplinter/src/run/network.rs]: How embarrassing. I removed this comment and will make a note to add this to the list of things we need to consider prior to stabilization.

**Comment by vaporos** [comment]
Fixed all the instances of old-copyright-header errors.

**Review by peterschwarz — APPROVED** [review]

**Review by agunde406 — APPROVED** [review]

**Pull request merged**